### ISSUE TITLE
Bootloop after upgrading to Sonoma 14.4

### ISSUE THREAD

**Issue by woody4165** [zearp/Nucintosh#151]
Hi all

First of all, thanks for sharing this repo on Github

I have a NUC nuc8i3beh with  i3-8109U CPU @ 3.00GHz, 16GB ram and SSD and NVME installed

I have installed from your repo Ventura and then upgraded easily to Sonoma, just changing to AirportItlwm.kext-Sonoma

It was going really well until yesterday, when I decided to update to Sonoma 14.4

After upgrading, the NUC restarted some times and booted correctly, so I've used, the first time, without no errors.

Now when I start using my user, but also with guest, I get a boot loop

I have also installed Windows 11 from the EFI shell and it's working fine.

Since I have no access to macos, what can I do?

I don't have any backup, is there a way to revert back or to find some solution?

Thanks

**Comment by zearp** [comment]
I had a bootloop using a beta of 14.4 but the final release installed fine. You may have to update the whole EFI not just the Intel wireless driver. the annoying part with boot loops is getting some details. What I do is record the screen at at least 60fps with my phone and then hope I can see the actual panic or error right before it reboots and a bit before that too.

You don't need to have access to your macOS install, you don't even need access to the EFI on your disk. You can format any usb stick as FAT32 in Windows and create an EFI folder on there and boot from that into macOS. I suggest you do that if you can't upgrade your current EFI to the latest release. Also try to catch the actual errors if they persists after updating.

I can't do much unless I can see the errors and ideally reproduce your issue on one of my NUCs. Assuming yours is completely stock, no Broadcom or any other additional hardware. Let me know how it goes please!

**Comment by zearp** [comment]
I had a bootloop using a beta of 14.4 but the final release installed fine. You may have to update the whole EFI not just the Intel wireless driver. The annoying part with boot loops is getting some details, even with the right boot parameters it can still just reboot instead of halting and showing you the error.

What I do is record the screen at at least 60fps with my phone and then hope I can see the actual panic or error right before it reboots and a bit before that too.

You don't need to have access to your macOS install, you don't even need access to the EFI on your disk. You can format any usb stick as FAT32 in Windows and create an EFI folder on there and boot from that into macOS. I suggest you do that if you can't upgrade your current EFI to the latest release. Also try to catch the actual errors if they persists after updating.

I can't do much unless I can see the errors and ideally reproduce your issue on one of my NUCs. Assuming yours is completely stock, no Broadcom or any other additional hardware. Let me know how it goes please!

**Comment by zearp** [comment]
Oh come to think of it, the Intel wireless driver might be the issue. I don't think I have that enabled on my Sonoma test machine. Try disabling that one in the config too, just for good measure. I have a backup on 14.3.1 or something that I can try to upgrade. but I'll wait for your error message and/or testing results before doing that.

**Comment by zearp** [comment]
I also had some boot loops, I guess it also creates a weird issue where whilst updating when the kext is either too new or too old. I downloaded the update as normal and when it's done downloading and preparing the update to quickly mount the EFI and disable the Intel wireless kext. Not needed when using wired of course. Then let it update and once it's running 14.4 replace the kext with the 14.4 version and reboot again to enable wifi.

**Comment by zearp** [comment]
I also had some boot loops, I guess it also creates a weird issue where whilst updating the kext is either too new or too old. I downloaded the update as normal and when it's done downloading and preparing the update to quickly mount the EFI and disable the Intel wireless kext. Not needed when using wired of course. Then let it update and once it's running 14.4 replace the kext with the 14.4 version and reboot again to enable wifi.

**Comment by zearp** [comment]
Another upgrade method would be to create a full 14.4 installer using [this](https://swcdn.apple.com/content/downloads/47/20/052-62010-A_IGTQLDJJXD/vu00gjm2kgj82xrpy4z05zgy1hm2e7in1x/InstallAssistant.pkg), but you'd need access to macOS for that. You can always install the same or newer version on top of an already installed macOS without losing any settings. IF you have a backup a clean install and then using the migration tool will also do the trick and saves a lot of time.

Putting this extra info here as I suspect others will run into this boot loop too. This is one of the first times an update really messes something up.

**Comment by woody4165** [comment]
Thanks @zearp 

When I boot the NUC I get the login page and then the bootloop

I have another 1TB SSD where is located my previous macos, BigSur that I luckly have still untouched and I would like to preserve...

I have the USB stick created following your instructions, with Ventura. If I start from it I get the procedure to install new maos.
Can I use that USB stick in some way?

Otherwise, after formatting a new USB stick in Windows with FAT32, what should I do?

Unfortunately, I don't have another 1TB SSD free to do a clone, I have only 500GB

Here is a "slowmotion" video that I made when booting, but after login, I get no error.

https://drive.google.com/file/d/1EjlhbSNqU_VKUKM1y51V9tadZnT1YNPS/view?usp=drive_link

Thanks

**Comment by woody4165** [comment]
After I fix this issue, in some way I have to solve it, how should I make a bootable USB sitck that I can use in such situation ?

Thanks again

**Comment by zearp** [comment]
I don't have permission to view the video. But if you haven't deleted the macOS partition on your disk you can boot into it using the method I described above. Format a usb stick to fat32 in Windows and copy the EFI to there then reboot and using the boot menu of your bios boot from the usb stick, it should load OpenCore and allow you to boot into macOS. Make sure to disable AirportItlwm.kext in the config by the setting enabled to false.

Once back in macOS I would suggest downloading the full 14.4 installer (see link above) and then create a new install usb stick with that or install it from within macOS. The former should work the latter might not always work. I had to use the full installer usb installing on top of macOS method to sort mine out. Somehow the thing the updater boots into always broke and showed some graphical error despite verbose boot being enabled. Guess there are some deep changed in 14.4!

tl;dr
Create a new full installer and install 14.4 with the latest release EFI and install it on top of an existing macOS or fresh and restore apps and settings from a backup using the migration tool (the installer will offer this when doing a fresh install).

**Comment by zearp** [comment]
I made a full installer using the latest release EFI and I can sue wireless inside the installer but upgrading still results in a bad time (checkpoint errors and a boot loop). Will format the disks and try a fresh install, see what that does.

**Comment by zearp** [comment]
I made a full installer using the latest release EFI and I can use wireless inside the installer but upgrading still results in a bad time (checkpoint errors and a boot loop). Will format the disks and try a fresh install, see what that does. Setting secure boot model to disabled may work too. I have not tested that.

**Comment by zearp** [comment]
Setting secure boot model to Disabled seems to do have done the trick. No idea why this works but it will do for now. I don't have much time to investigate this weekend.

So updating the Intel wireless driver and setting secure boot model to disabled is whats needed to get on 14.4 😅

**Comment by woody4165** [comment]
So, now I have created an USB stick in FAT32 from Windows and I copy the EFI folder directly from your project into it.

I booted, but in the EFI start menu there was no new item.
I selected Sonoma and this time did boot.

I don't it used the EFI on the USB stick, but I will see next time I boot without it



I see in the status bar that the Wifi icon has an exclamation mark, but I don't use Wifi.

I've downloaded the AirportItlwm.kext-Sonoma that you linked some messages above and now I have to replace in my EFI folder.
I use OpenCore Configurator, then I have to go to Kernel, do a Scan/Browse and save the config.plist again.

Is it correct?

Thanks

**Comment by zearp** [comment]
Please follow my instructions and the ones on the readme.md, I can't help you when using tools like OCC. I've upgraded successfully to 14.4 using the latest release EFI on the repo.

![Screenshot 2024-03-09 at 20 54 45](https://github.com/zearp/Nucintosh/assets/63272687/374b6db2-a08f-49ed-ad6c-f087441f5a81)

**Comment by woody4165** [comment]
This is mine now

![Screenshot 2024-03-09 alle 21 15 43](https://github.com/zearp/Nucintosh/assets/6501444/a77963e5-1aed-4b9e-b8c5-5f8c3d875f45)


If I follow the instructions to create the USB stick, when I start from it, it will show the install procedure for Sonoma, in this case.

Is there a way to create an USB stick with the EFI folder from your repo (or eventually from my actual EFI folder) and have a USB stick that will start even if there is something wrong in the SSD main EFI ?

BTW, now it's working fine after several reboot

Thanks

**Comment by zearp** [comment]
I’m sorry but I’m not going to answer the same question for a third time.
